tests/RecordTests/FunctionalTest/k4a_cpp_ft.cpp

```diff
@@ -0,0 +1,511 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <k4arecord/record.hpp>
+#include <k4arecord/playback.hpp>
+#include <utcommon.h>
+
+#include <gtest/gtest.h>
+
+using namespace k4a;
+
+const std::string MKV_FILE_NAME("./k4a_cpp_ft.mkv");
+
+int main(int argc, char **argv)
+{
+    return k4a_test_common_main(argc, argv);
+}
+
+class k4a_cpp_ft : public ::testing::Test
+{
+public:
+    virtual void SetUp()
+    {
+        std::remove("*.mkv"); // remove old test files in the event of a crash
```

Comment:
>std::remove("*.mkv") [](start = 8, length = 20)

I'm pretty sure this doesn't actually work - I don't think wildcards are supported using std::remove.  I think you need to call std::remove on every file you want to delete.  Unfortunately, std::filesystem is a C++17 feature, and that's the only built-in portable way to enumerate files I'm aware of, so I don't think we have access to the stuff to enumerate files in the K4A project (it's on C++11, I think?), so you might just have to hardcode them here :(